Core/Src/main.c

```diff
@@ -87,16 +91,41 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
 
+  const int MAX_ADC_VAL = 1024;
+  uint8_t dataIn[3] = {0};
+  uint8_t dataOut[3] = {0b00000001, 0b10000000, 0b00000000};
```

Comment:
if this value is not going to change it should be marked as CONST... if you do this, please follow the ALL_CAPS convention